packages/core/src/provider.ts

```diff
@@ -339,14 +341,16 @@ export async function getBalance(
 ): Promise<Balances | null> {
   // chain we don't recognize and don't have a rpcUrl for requests
   if (!chain) return null
-
-  const provider = getProvider(chain)
+  
+  const { wallets } = state.get()
 
   try {
-    const balanceWei = await provider.getBalance(address)
-
+    const wallet = wallets.find(wallet => wallet.label)
```

Comment:
@mahmud-bn does this just return the first wallet that has the label property? 
also what happens if `wallet` is `undefined`? 
maybe to make it more explicit do something like:
```
const wallet = wallets.find(wallet => !!wallet.label)
```
or even?
```
const wallet = wallets.find(wallet => !!wallet)
```

Reply:
Or even could do it on the provider which would also check that the provider is defined:
```
const wallet = wallets.find(wallet => !!wallet.provider)
```

Reply:
@taylorjdawson 
- Yes, it does. Calling the getBalance() function gets the updated balance of all connected wallets.
- Since getBalance() is called after the wallet connection has taken place, I assume that was enough protection. But I've added the provider check suggestion.